pkg/console/subresource/deployment/deployment.go

```diff
@@ -184,6 +211,64 @@ func LogDeploymentAnnotationChanges(client appsclientv1.DeploymentsGetter, updat
 	}
 }
 
+func containerResourceRequirements(cpu, memory string) corev1.ResourceRequirements {
+	return corev1.ResourceRequirements{
+		Requests: map[corev1.ResourceName]resource.Quantity{
+			corev1.ResourceCPU:    resource.MustParse(cpu),
+			corev1.ResourceMemory: resource.MustParse(memory),
+		},
+	}
+}
+
+func tolerations() []corev1.Toleration {
+	return []corev1.Toleration{
+		{
+			Key:      "node-role.kubernetes.io/master",
+			Operator: corev1.TolerationOpExists,
+			Effect:   corev1.TaintEffectNoSchedule,
+		},
+		{
+			Key:               "node.kubernetes.io/unreachable",
+			Operator:          corev1.TolerationOpExists,
+			Effect:            corev1.TaintEffectNoExecute,
+			TolerationSeconds: &tolerationSeconds,
+		},
+		{
+			Key:               "node.kubernetes.io/not-reachable",
+			Operator:          corev1.TolerationOpExists,
+			Effect:            corev1.TaintEffectNoExecute,
+			TolerationSeconds: &tolerationSeconds,
+		},
+	}
+}
+
+func affinity(infrastructureConfig *configv1.Infrastructure) *corev1.Affinity {
+	if infrastructureConfig.Status.InfrastructureTopology == configv1.SingleReplicaTopologyMode {
```

Comment:
I believe the two deployments should be looking at different properties. Since console itself has a label selector for master nodes, it should look at `controlPlaneTopology`. The downloads deployment runs on any Linux node, so it should look for `infrastructureTopology`.

Reply:
@spadgett  update, but thinking if it makes sense to keep the anti-affinity rules for the `downloads` deployment?